src/components/signup/SignupState.js

```diff
@@ -109,18 +107,19 @@ const Signup = ({ navigation }: { navigation: any, screenProps: any }) => {
   const [unrecoverableError, setUnrecoverableError] = useState(null)
   const [ready, setReady]: [Ready, ((Ready => Ready) | Ready) => void] = useState()
   const [state, setState] = useState(initialState)
-  const [loading, setLoading] = useState(false)
   const [countryCode, setCountryCode] = useState(undefined)
   const [createError, setCreateError] = useState(false)
-  const [showNavBarGoBackButton, setShowNavBarGoBackButton] = useState(true)
   const [finishedPromise, setFinishedPromise] = useState(undefined)
-  const [title, setTitle] = useState('Sign Up')
   const [, hideDialog, showErrorDialog] = useDialog()
   const shouldGrow = store.get && !store.get('isMobileSafariKeyboardShown')
 
+  const { preparing: walletPreparing, success: signupSuccess, setWalletPreparing, setSuccessfull } = useContext(
+    AuthContext,
+  )
+
   const navigateWithFocus = (routeKey: string) => {
     navigation.navigate(routeKey)
-    setLoading(false)
+    setWalletPreparing(false)
```

Comment:
all of the setWalletPreparing looks incorrect to me, either the logic or the naming.
the name suggest this is about the modal we show once user is login in with Torus. "We are preparing you wallet...".
During the signup process the setLoading is showing a spinner when something happens, ie sending email/sms, verifying the code etc...

Reply:
@sirpy fixed